lib/fetching/bot-master.js

```diff
@@ -116,6 +117,12 @@ BotMaster.prototype._addSourceListeners = function(emitter) {
     var bot = self._getBot(source._id);
     if (bot) bot.stop();
   });
+
+  // Listen to `disable` event from the fetching process
+  // This meant for non-recoverable errors coming from the sources/bots
+  Source.schema.on('source:disable', function(source) {
+    self._getBot(source._id).stop();
+  });
```

Comment:
I think this belongs somewhere else: this isn't really part of `_addSourceListeners`. Let's pull this out into like `BotMaster.addOwnListeners` and call that from `lib/fetching.js`